File: Parking Lot/Parking Lot C#/Model/Slot.cs
namespace Model
{
    public class Slot(int id)
    {

        public Car ParkedCar { get; set;}
        public int Id { get; set; } = id;
        public bool IsSlotFree()
        {
            return ParkedCar == null;
        }
        public void AssignCar(Car car)
        {
            ParkedCar = car;
        }
        public void UnassignCar()
        {
            ParkedCar = null;
        }
    }
}

File: Parking Lot/Parking Lot Java/src/main/java/org/example/models/Slot.java
package org.example.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.Objects;

@Getter
public class Slot {

    private Integer id;
    private Car car;

    public Slot(Integer id) {
        this.id = id;
    }

    public void assignCar(Car car) {
        this.car = car;
    }

    public void unassignCar() {
        this.car = null;
    }

    public boolean isSlotFree() {
        return Objects.isNull(this.car);
    }
}
